packages/core/src/browser/fetchObservable.ts

```diff
@@ -93,25 +94,63 @@ function afterSend(
   responsePromise: Promise<Response>,
   startContext: FetchStartContext
 ) {
-  const reportFetch = (response: Response | Error) => {
+  const reportFetch = (response: Response | Error, resolveDuration?: Duration) => {
     const context = startContext as unknown as FetchCompleteContext
     context.state = 'complete'
     context.duration = elapsed(context.startClocks.timeStamp, timeStampNow())
-
+    context.resolveDuration = resolveDuration
     if ('stack' in response || response instanceof Error) {
       context.status = 0
       context.isAborted = response instanceof DOMException && response.code === DOMException.ABORT_ERR
       context.error = response
-
-      observable.notify(context)
     } else if ('status' in response) {
       context.response = response
       context.responseType = response.type
       context.status = response.status
       context.isAborted = false
-
-      observable.notify(context)
     }
+    observable.notify(context)
   }
-  responsePromise.then(monitor(reportFetch), monitor(reportFetch))
+
+  const waitForResponseToFinish = (response: Response): Promise<Duration> =>
+    new Promise((resolve) => {
+      const resolveDuration = elapsed(startContext.startClocks.timeStamp, timeStampNow())
+
+      const responseClone = response.clone()
+      const reader = responseClone.body?.getReader()
+
+      if (reader && ReadableStream) {
+        new ReadableStream({
+          start(controller) {
+            return pump()
+
+            function pump(): Promise<void> {
+              return (reader as ReadableStreamDefaultReader<Uint8Array>).read().then(
+                ({ done }) => {
+                  if (!done) return pump()
+                  controller.close()
+                  resolve(resolveDuration)
+                },
+                () => {
+                  controller.close()
+                  resolve(resolveDuration)
+                }
+              )
+            }
+          },
+        })
+      } else {
+        resolve(resolveDuration)
+      }
+    })
+
+  responsePromise.then(
+    monitor((response) =>
+      waitForResponseToFinish(response).then(
+        (resolveDuration: Duration) => reportFetch(response, resolveDuration),
+        () => reportFetch(response)
+      )
+    ),
+    monitor(reportFetch)
+  )
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** Since we are already reading responses for logs network error collection, there are probably some thoughts to have on that front.

- Do we really need to wait for response to finish for all fetch instrumentation usages? 
- Should we mutualise some things with our current [response reading code](https://github.com/DataDog/browser-sdk/blob/main/packages/logs/src/domain/logsCollection/networkError/networkErrorCollection.ts#L201)?

Reply:
> Should we mutualise some things with our current response reading code?
Yes agreed

> Do we really need to wait for response to finish for all fetch instrumentation usages?
I agree this isn't ideal, however if we wanted to finish earlier we cannot determine a precise duration. Happy to sync up about this issue